extensions/common/context.cc

```diff
@@ -294,6 +294,27 @@ void populateHTTPRequestInfo(bool outbound, bool use_host_header_fallback,
   }
 }
 
+void populateExtendedHTTPRequestInfo(RequestInfo* request_info) {
+  getValue({"source", "address"}, &request_info->source_address);
+  getValue({"destination", "address"}, &request_info->destination_address);
+
+  getValue({"request", "referer"}, &request_info->referer);
+  getValue({"request", "user_agent"}, &request_info->user_agent);
+  getValue({"request", "id"}, &request_info->request_id);
+  std::string trace_sampled;
+  if (getValue({"request", "headers", "x-b3-sampled"}, &trace_sampled) &&
+      !trace_sampled.empty() && trace_sampled != "0") {
```

Comment:
`trace_sampled != "0"` seems weird to me

Reply:
It is a string, and you can disable sampling by setting it to “0”

Reply:
Though a nit: we could just check for ==“1” and not worry about empty.

Reply:
https://github.com/istio/proxy/pull/2627